server-apollo/index.js

```diff
@@ -1,5 +1,15 @@
 const { ApolloServer, gql } = require('apollo-server');
+const { GraphQLError } = require('graphql')
 
+class ConflictError extends GraphQLError {
+    constructor(type, data){
+        const message = "conflict has occured"
+        super(message)
+        this.data = data
+        this.type = type
```

Comment:
If this is `ConflictError` then type makes no sense. Type can be used as string to check error on client.
I guess we may need just have something like AGSyncError or keep each error as separate class. 
I have no strong opinion or preference.